pkg/server/test/list_objects.go

```diff
@@ -158,6 +161,33 @@ func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore
 			minimumResultsExpected: 1,
 			allResults:             []string{"team:1"},
 		},
+		{
+			// should not return any useful data as it times out
+			// however, there should not be any error
+			name:   "list_objects_timeout",
+			schema: typesystem.SchemaVersion1_1,
+			model: `
+			type user
+			type repo
+			  relations
+				define admin: [user] as self
+			`,
+			tuples: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("repo:1", "admin", "user:alice"),
+				tuple.NewTupleKey("repo:2", "admin", "user:alice"),
+			},
+			user:       "user:alice",
+			objectType: "repo",
+			relation:   "admin",
+			contextualTuples: &openfgapb.ContextualTupleKeys{
+				TupleKeys: []*openfgapb.TupleKey{tuple.NewTupleKey("repo:3", "admin", "user:alice")},
+			},
+			maxResults:             2,
+			minimumResultsExpected: 0,
+			allResults:             []string{},
```

Comment:
`allResults` should be the list of results that you'd expect if you ran `Check` for all objects, without caring about the ListObjects API settings. So in this test `repo:1 , repo:2, repo:3` are all the valid results.

The expectations that we have for this test is that:
-  `err == nil`
- `0 <= len(results)`


If so, `minimumResultsExpected: 0` is okay, but `maxResults` can be set to a million and it should still pass. Perhaps we should have a default of 100000 and just not set it in this test

Reply:
No, we specifically expect there will be no objects returned as all reads are timed out

Reply:
I will add comments to make it explicit that we expect empty array and test consider failure if there are any.

Reply:
> we specifically expect there will be no objects returned as all reads are timed out.

This is checked via `minimumResultsExpected: 0`. You should still set `allResults = [repo:1 , repo:2, repo:3]` because this assertion will still hold true:

```
require.Subset(t, test.allResults, res.Objects)
```